src/adapter/XmlSelectorAdapter.js

```diff
@@ -0,0 +1,22 @@
+const xpath = require('xpath');
+const { XMLSerializer, DOMParser } = require('xmldom');
+const DataSelectorPort = require('../domain/common/port/DataSelectorPort');
+
+class XmlSelectorAdapter extends DataSelectorPort {
+  canHandle(contentType) {
+    return ['xml', 'html'].includes(contentType);
+  }
+
+  select(data, path) {
+    const doc = (new DOMParser()).parseFromString(data);
```

Comment:
Should `DOMParser` and `XLMSerializer` be always created for each select operation or could we share them as instance variables and create them only once in the constructor?